com.unity.netcode.gameobjects/Runtime/Serialization/BufferSerializer.cs

```diff
@@ -0,0 +1,313 @@
+        /// </summary>
+        /// <param name="value">Value to serialize</param>
+        public void SerializeNetworkSerializable<T>(ref T value) where T : INetworkSerializable
+        {
+            m_Implementation.SerializeNetworkSerializable(ref value);
+        }
+
+        /// <summary>
+        /// Serialize an INetworkSerializable
+        /// 
+        /// Throws OverflowException if the end of the buffer has been reached.
+        /// Write buffers will grow up to the maximum allowable message size before throwing OverflowException.
+        /// </summary>
+        /// <param name="value">Value to serialize</param>
+        public void SerializeValue(ref INetworkSerializable value)
+        {
+            m_Implementation.SerializeValue(ref value);
+        }
+
+        /// <summary>
+        /// Serialize a GameObject
+        /// 
+        /// Throws OverflowException if the end of the buffer has been reached.
+        /// Write buffers will grow up to the maximum allowable message size before throwing OverflowException.
+        /// </summary>
+        /// <param name="value">Value to serialize</param>
+        public void SerializeValue(ref GameObject value)
+        {
+            m_Implementation.SerializeValue(ref value);
+        }
+
+        /// <summary>
+        /// Serialize a NetworkObject
+        /// 
+        /// Throws OverflowException if the end of the buffer has been reached.
+        /// Write buffers will grow up to the maximum allowable message size before throwing OverflowException.
+        /// </summary>
+        /// <param name="value">Value to serialize</param>
+        public void SerializeValue(ref NetworkObject value)
+        {
+            m_Implementation.SerializeValue(ref value);
+        }
+
+        /// <summary>
+        /// Serialize a NetworkBehaviour
+        /// 
+        /// Throws OverflowException if the end of the buffer has been reached.
+        /// Write buffers will grow up to the maximum allowable message size before throwing OverflowException.
+        /// </summary>
+        /// <param name="value">Value to serialize</param>
+        public void SerializeValue(ref NetworkBehaviour value)
+        {
+            m_Implementation.SerializeValue(ref value);
+        }
+
+        /// <summary>
+        /// Serialize a string.
+        ///
+        /// Note: Will ALWAYS allocate a new string when reading.
+        /// 
+        /// Throws OverflowException if the end of the buffer has been reached.
+        /// Write buffers will grow up to the maximum allowable message size before throwing OverflowException.
+        /// </summary>
+        /// <param name="s">Value to serialize</param>
+        /// <param name="oneByteChars">
+        /// If true, will truncate each char to one byte.
+        /// This is slower than two-byte chars, but uses less bandwidth.
+        /// </param>
+        public void SerializeValue(ref string s, bool oneByteChars = false)
```

Comment:
I've been adding some stuff that fits this need in the INetworkMessage branch because I ran into the need for it there, so I'm going to resolve this and it'll be addressed there.